src/widgets/barchart.rs

```diff
@@ -335,6 +335,31 @@ impl<'a> BarChart<'a> {
     }
 
     fn render_horizontal_bars(self, buf: &mut Buffer, bars_area: Rect, max: u64) {
+        // get the longest label
+        let label_size = self
+            .data
+            .iter()
+            .map(|group| {
+                group
+                    .bars
+                    .iter()
+                    .map(|bar| bar.label.clone().map_or(0, |label| label.width()))
+                    .max()
+                    .unwrap_or(0)
+            })
+            .max()
+            .unwrap_or(0) as u16;
+
+        let label_x = bars_area.x;
+        let bars_area = {
+            let margin = if label_size == 0 { 0 } else { 1 };
+            Rect {
+                x: bars_area.x + label_size + margin,
+                width: bars_area.width - label_size - margin,
+                ..bars_area
+            }
```

Comment:
I find myself often writing this sort of logic for a Rect in apps and code and have been trying to think of an ergonomic method name for it - what would you think about this being `Rect::inset(x, y)` (where `Rect::offset(x, y)` would not change the width and height?

Reply:
If I understand correctly, this would end up to something like that.

```rust
let newRect = rect.inset(label_size + margin, 0);
```

Seems very useful and less error-prone. Naming looks good to me.